min_heap_binary_tree.rb

```diff
@@ -80,4 +101,68 @@ def parent(index)
   def parents_index(index)
     index / 2
   end
+
+  def left_child_index(index)
+    index * 2
+  end
+
+  def right_child_index(index)
+    (index * 2) + 1
+  end
+
+  def index_to_swap_with_two_children(index)
+    if right_child(index).value <= left_child(index).value
+      (2 * index) + 1
+    else
+      2 * index
+    end
+  end
+
+  def only_left_child_smaller?(index)
+    (left_child(index).value < tree[index].value) && (tree[index].value <= right_child(index).value)
+  end
+
+  def only_right_child_smaller?(index)
+    (tree[index].value < left_child(index).value) && (right_child(index).value < tree[index].value)
+  end
+
+  def left_child_larger?(index)
+    left_child(index) != nil && tree[index].value > left_child(index).value
+  end
+
+  def both_children_smaller_swap(replacement_index)
+    temp = index_to_swap_with_two_children(replacement_index)
```

Comment:
What would be a different name for this variable other than `temp`?

And spacing after it.